### ISSUE TITLE
Android Boot splash regressions with background color and boot up time

### ISSUE THREAD

**Issue by georgwacker** [godotengine/godot#85264]
### Godot version

4.2rc1, 4.1.3, 4.1.2

### System information

Godot v4.2.rc1 - macOS 14.1.1 - Vulkan (Mobile) - dedicated AMD Radeon RX 6800

### Issue description

On 4.2 RC1 the boot splash with a white background flashes for a split second, then shows a black screen for 6-7 seconds (timed), then shows the initial screen of the game.

On 4.1.3 the colored boot splash shows for about 1 second, then a 4 second black screen, then the initial game screen. 

On 4.1.2 the colored boot splash shows for about 1 second, then a 2.5 seconds black screen, then the initial game screen.

To summarize:
- 4.1.2 and 4.1.3 used the background color set via `application/boot_splash/bg_color` for the boot splash, 4.2 RC1 does not anymore.
- Boot up time has gotten worse since 4.1.2

Could this be related to #76821?

**Expected behavior** for the Android boot splash in my eyes: show colored boot splash for loading duration, then transition into initial game screen seamlessly without a black screen.  

### Steps to reproduce

Export to Android with Forward Mobile, run via USB debug

### Minimal reproduction project

N/A

**Comment by akien-mga** [comment]
Given that the 4.2 release is imminent, I'm flagging this for 4.3 as I doubt we'll have time to debug, fix and test a solution rapidly enough without compromising the stability. It seems like the situation went from bad to worse so both the regression and the initial 4.1 problem should likely be addressed together to finally fix the boot splash issues on Android.

**Comment by georgwacker** [comment]
That sounds reasonable, I've also added the expected behavior in the description for reference. 

Personally I would like to see an option to either show the android style boot splash (with app icon) or a custom boot splash with a publisher/developer logo (like we get on other platforms).

**Comment by akien-mga** [comment]
Fixed by #92965.
